src/SandboxCoordinator.ts

```diff
@@ -60,12 +60,13 @@ export default class SandboxCoordinator {
   }
 
   private createSandboxes(): Promise<Sandbox[]> {
-    const cpuCount = os.cpus().length;
+    const numConcurrentRunners = this.options.maxConcurrentTestRunners || os.cpus().length;
```

Comment:
Ok done and added tests for the additional cases.

Let me know if it's all good and I'll squash the commits.